src/components/createInstance.js

```diff
@@ -0,0 +1,202 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import {
+  Container,
+  Breadcrumb,
+  Header,
+  Form,
+  Button,
+  Icon
+} from 'semantic-ui-react'
+import css from './css/approveRequest.css'
+import ErrorBoundary from './error-boundary'
+import { createFilemanager } from '../actions/itemActions'
+import { ONE_GB } from '../constants'
+
+const spaceOptions = [
+  { key: '1', text: '1 GB', value: 1 * ONE_GB },
+  { key: '2', text: '2 GB', value: 2 * ONE_GB },
+  { key: '3', text: '5 GB', value: 5 * ONE_GB },
+  { key: '4', text: '10 GB', value: 10 * ONE_GB }
+]
+
+const roleOptions = [
+  { key: '1', text: 'Student', value: 'Student' },
+  { key: '2', text: 'FacultyMember', value: 'FacultyMember' },
+  { key: '3', text: 'Maintainer', value: 'Maintainer' },
+  { key: '4', text: 'Guest', value: 'Guest' }
+]
+
+class CreateInstance extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      filemanagerName: '',
+      rootFolderName: '',
+      rolesAllowed: [],
```

Comment:
try to make fields in one objects and set that object as a single state.(see FolderFormModal for reference)